tests/test_benchmark.py

```diff
@@ -423,6 +418,27 @@ def setUpClass(self):
     Building.COMPILER = CLANG
     Building.COMPILER_TEST_OPTS = [OPTIMIZATIONS]
 
+  # avoid depending on argument reception from the commandline
+  def hardcode_arguments(self, code):
+    if not code:
+      return code
+    main_pattern = 'int main(int argc, char **argv)'
+    assert main_pattern in code
+    code = code.replace(main_pattern, 'int benchmark_main(int argc, char **argv)')
+    code += '''
+      int main() {
+        // use volatiles so the compiler doesn't optimize all the work away
+        volatile int argc = 2;
+        typedef char** charStarStar;
+        volatile charStarStar argv;
+        argv[0] = (char*)"./program.exe";
+        argv[1] = (char*)"%s";
```

Comment:
This looks like its shouldn't work since you are not allocating any memory for argv.
Shouldn't it be more like:
```
char newArgv[] = { (char*)"./program.exe", (char*)"%s", NULL }
int newArgc = 2;
```
I can't see why you would want to use volatile, or shadow the input args.

Reply:
Thanks, yeah, this was not allocating memory... fixed.

I was thinking volatile would prevent any chance of the compiler doing fancy LTO that depends on the argument, but yeah, I guess that's overkill. Removed.